internal/pkg/task/task.go

```diff
@@ -44,3 +52,15 @@ var (
 func taskFamilyName(groupName string) string {
 	return fmt.Sprintf(fmtTaskFamilyName, groupName)
 }
+
+func tasks(ecsTasks []*ecs.Task) []*Task {
```

Comment:
What do think of this:

```go
func (t *Task) fromECS(ecsTask *ecs.Task) {
   t.TaskARN = aws.StringValue(task.TaskArn)
   t.ClusterARN = aws.StringValue(task.ClusterArn)
   t.StartedAt = aws.TimeValue(task.StartedAt)
}

func convertECSTasks(ecsTasks []*ecs.Task) []*Task {
  tasks := make([]*Task, len(ecsTasks))
  for idx, ecsTask := range ecsTasks {
     tasks[idx].fromECS(ecsTask)
  }
  return tasks
}
```

Reply:
This probably won't work because `fromECS` attempts to change the value of the pointer, not the object that the pointer points to. It would be fine if the initialization is done in `convertECSTasks`

Reply:
derp yeah you're right, since the receiver is copied so it doesn't change the actual pointer :( 

Ok scratch that then :)